common/src/main/java/org/conscrypt/AbstractSessionContext.java

```diff
@@ -76,7 +75,7 @@ protected boolean removeEldestEntry(
         // Make a copy of the IDs.
         final Iterator<NativeSslSession> iter;
         synchronized (sessions) {
-            iter = Arrays.asList(sessions.values().toArray(new NativeSslSession[sessions.size()]))
+            iter = Arrays.asList(sessions.values().toArray(new NativeSslSession[0]))
```

Comment:
It is optimisation in OpenJDK [1], but it is true for Android?

[1] https://shipilev.net/blog/2016/arrays-wisdom-ancients/

Reply:
On android the difference is miniscule (<2% even on very large lists). See b/183484607

Reply:
Thanks for taking the time to benchmark this, Alex!  None of these code paths should be hot either, but good to know it shouldn't make a significant difference.